src/appshell/internal/appshellconfiguration.cpp

```diff
@@ -22,7 +22,16 @@ using namespace mu::appshell;
 
 bool AppShellConfiguration::isAppUpdatable() const
 {
-#if defined(APP_UPDATABLE)
+#ifdef APP_UPDATABLE
+    return true;
+#else
+    return false;
+#endif
+}
+
+bool AppShellConfiguration::isFullScreenAvailable() const
+{
+#ifndef Q_OS_MAC
     return true;
 #else
     return false;
```

Comment:
But full screen is also available on over platfroms...

Reply:
You misread this: it's `#ifndef`, not `#ifdef`. This means fullscreen is available on every platform *except* macos. 
I didn't know macos had no fullscreen mode though...

Reply:
But macOS _does_ have full screen! 
![image](https://user-images.githubusercontent.com/48658420/108635344-4c9da300-747f-11eb-9b1d-9c3e52002639.gif)
The default keyboard shortcut is ⌃⌘F.

Reply:
I don't know why it's there then... Maybe an old qt bug? Maybe something can be found in git history?

Reply:
I added that a while ago. Whatever version of macOS was current then automatically appended the option to the View menu even if it was already present, so we hid the first one in order not to duplicate the macOS-provided one.

Reply:
@IsaacWeiss Thanks, that's good to know. For me on macOS Big Sur, no Full Screen option is added to the menu, neither in MuseScore 3 nor in 4, so now, there is no such option in the menu at all. However, the keyboard shortcut magically works. 
But indeed, on macOS 10.15, a full screen item is added. So we would need a version check, which is doable but not really nice. Another disadvantage of the macOS provided item is that it seems to use the system language instead of MuseScore's language. 

But maybe there is good news: there seems to be a way to stop macOS from adding that item, see https://stackoverflow.com/questions/52154977/how-to-get-rid-of-enter-full-screen-menu-item. We could put it in the `init` method of MacOSPlatformTheme. This gives us full control, but also gives us the responsibility to manage full screen correctly. We would need to see how that works out in practice.

Reply:
Not magic, see #2522. :-) It is managed correctly—the two menu entries trigger(ed) the exact same action.

Reply:
Ah, nice! I didn't realize for a moment that keyboard shortcuts of course also work when they are not in a menu item. In that case, we can indeed be confident about abandoning macOS's full screen item and use our own, because in fact that is already kind of what happens. I could create a PR soon.
@Eism If we decide to do it like that, you could actually remove the whole `isFullScreenAvailable` method, as it should be always true.

We could think a bit about "Enter" and "Exit". It would be nice if actions could have multiple titles based on the current state. For example, "Show Palettes" and "Hide Palettes" instead of only "Toggle Palettes". It seems that other macOS apps also prefer that. With our current actions system, that isn't possible, and it would maybe involve many changes.

Reply:
Thanks guys for getting involved in this issue
I really didn't know why fullscreen didn't work on macOS and decided not to change the logic. A little later, I wanted to investigate this problem
Ok, now I understand(thanks to your help) that I can safely remove this code

Reply:
thanks, removed